From a111f965d46354a502484cc525b740e83093da5d Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Fri, 14 Aug 2020 18:14:46 +0200
Subject: [PATCH] ifupdown: enable udhcpc backgrounding HYP-17009

The former default behaviour was to exit with failure if lease is not
immediately obtained. This results in permanent network disconnect if
DHCP server is (accidentally) not available at boot time. :(

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Signed-off-by: Steffen Sledz <sledz@dresearch-fe.de>
---
 inet.defn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inet.defn b/inet.defn
index 5c16fab..3093396 100644
--- a/inet.defn
+++ b/inet.defn
@@ -103,7 +103,7 @@ method dhcp
         if (execable("/sbin/dhclient"))
     /sbin/pump -i %iface% [[-h %hostname%]] [[-l %leasehours%]] \
         elsif (execable("/sbin/pump"))
-    /sbin/udhcpc -n -p /run/udhcpc.%iface%.pid -i %iface% [[-x hostname:%hostname%]] \
+    systemd-run --quiet --service-type=forking --unit=hip-udhcpc_%iface% /sbin/udhcpc -b -p /run/udhcpc.%iface%.pid -i %iface% [[-x hostname:%hostname%]] \
         elsif (execable("/sbin/udhcpc"))
     /sbin/dhcpcd [[-h %hostname%]] [[-i %vendor%]] [[-I %client%]] \
            [[-l %leasetime%]] [[-m %metric%]] %iface% \
-- 
2.16.4

